src/filter-image/FilterImage.tsx

```diff
@@ -20,12 +29,17 @@ const getFilters = (filters: FilterImageProps['filters']) => {
 };
 
 export const FilterImage = (props: FilterImageProps) => {
-  const { source, style, ...imageProps } = props;
+  const { filters, source, style, ...imageProps } = props;
+  const hasFilter = filters.length > 0;
+  const filterId = useMemo(() => `RNSVG-${getRandomNumber()}`, [filters]);
```

Comment:
Do we need to change the `filterId` when the `filters` change?

Reply:
After some tests, we don't. Removed in https://github.com/software-mansion/react-native-svg/pull/2346/commits/af12b62af91bc39b87f1b82630eb08233793f657